x/rollup/tx/l1_data.go

```diff
@@ -0,0 +1,81 @@
+package tx
+
+import (
+	"fmt"
+	"math/big"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/rlp"
+	rollupkeeper "github.com/polymerdao/monomer/x/rollup/keeper"
+)
+
+// L1DataAnteHandler will consume gas to compensate the sequencer for posting
+// the transaction to Ethereum. The gas cost is calculated based on the Ecotone
+// upgrade and the sequencer is expected to post the transaction using blobs.
+func L1DataAnteHandler(ctx sdk.Context, tx sdk.Tx, rollupKeeper *rollupkeeper.Keeper) (sdk.Context, error) { //nolint:gocritic // hugeparam
+	if rollupKeeper == nil {
+		return ctx, errorsmod.Wrap(sdkerrors.ErrLogic, "rollup keeper is required for l1 data ante handler")
+	}
+
+	l1BlockInfo, err := rollupKeeper.GetL1BlockInfo(ctx)
+	if err != nil {
+		return ctx, fmt.Errorf("get l1 block info: %w", err)
+	}
+
+	baseFee := new(big.Int).SetBytes(l1BlockInfo.BaseFee)
+	baseFeeScalar := big.NewInt(int64(l1BlockInfo.BaseFeeScalar))
+	blobBaseFee := new(big.Int).SetBytes(l1BlockInfo.BlobBaseFee)
+	blobBaseFeeScalar := big.NewInt(int64(l1BlockInfo.BlobBaseFeeScalar))
+
+	l1GasUsed, err := getL1GasUsed(tx)
+	if err != nil {
+		return ctx, fmt.Errorf("get l1 gas used: %w", err)
+	}
+
+	const baseFeeMultiplier = 16
+	const divisor = 16000000 // 16e6
```

Comment:
nit: I think this should work.
```suggestion
	const divisor = 16e6
```

Reply:
I set `16e6` directly in the constant before but I just added a new [commit](https://github.com/polymerdao/monomer/pull/270/commits/db7bef55cc6bca3caa68ce8fc8eba8c906f2c4df) a couple hours ago to modify this to be an int instead since 16e6 defaults to a float64 type. Idt it makes a huge difference though so I'll go ahead and change it back